document-parsers/langchain4j-document-parser-apache-tika/src/test/java/dev/langchain4j/data/document/parser/apache/tika/ApacheTikaDocumentParserTest.java

```diff
@@ -34,14 +34,33 @@ void should_parse_doc_ppt_and_pdf_files(String fileName) {
         assertThat(document.metadata().toMap()).isEmpty();
     }
 
+    @ParameterizedTest
+    @ValueSource(strings = {
+            "test-file.doc",
+            "test-file.docx",
+            "test-file.ppt",
+            "test-file.pptx",
+            "test-file.pdf"
+    })
+    void should_parse_doc_ppt_and_pdf_files_with_metadata(String fileName) {
+
+        DocumentParser parser = new ApacheTikaDocumentParser(AutoDetectParser::new, null, null, null, true);
+        InputStream inputStream = getClass().getClassLoader().getResourceAsStream(fileName);
+
+        Document document = parser.parse(inputStream);
+
+        assertThat(document.text()).isEqualToIgnoringWhitespace("test content");
+        assertThat(document.metadata().toMap()).isNotEmpty();
```

Comment:
Is it possible to assert something about the metadata, e.g. filename?

Reply:
The filename is not part of the metadata provided by Tika. What is common in all files, is the Tika specific metadata for which class has parsed it. I added a check.